### PR TITLE
feat(core): add remove command

### LINKED ISSUE #41: Support `remove` command to remove a plugin that is currently installed by coffe
# Background

To run Coffe you are run the following command `cargo run -- --help` and you can run the following command to setup coffee for your core lightning node

```
cargo run -- setup <core lightning root path>
cargo run -- remote add lightningd https://github.com/lightningd/plugins.git
cargo run -- install <plugin_name>
```

Currently, there is no way to remove the plugin that we just installed, so this task required you to implement the remove command.

# Goal

Be able to run the following command 
```
cargo run -- remove <name_plugin>
```

### PR DESCRIPTION
**Pull request by tareknaser360**
fixes #41

### CODE REVIEW

File: coffee_cmd/src/main.rs [hunk 1]
```diff
@@ -38,7 +38,13 @@ async fn main() -> Result<(), CoffeeError> {
             }
             result
         }
-        CoffeeCommand::Remove => todo!(),
+        CoffeeCommand::Remove { plugin } => match coffee.remove(&plugin).await {
+            Ok(_val) => {
```
Comment (vincenzopalazzo) [inline coffee_cmd/src/main.rs:42]:
```suggestion
            Ok(_) => {
```

File: coffee_core/src/coffee.rs [hunk 2]
```diff
@@ -234,6 +234,28 @@ impl PluginManager for CoffeeManager {
         Err(err)
     }
 
+    async fn remove(&mut self, plugin: &str) -> Result<(), CoffeeError> {
```
Comment (vincenzopalazzo) [inline coffee_core/src/coffee.rs:237]:
```suggestion
    async fn remove(&mut self, plugin: &str) -> Result<Value, CoffeeError> {
```

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Small small comment but looks a lot better

File: coffee_core/src/coffee.rs [hunk 3]
```diff
@@ -234,6 +234,28 @@ impl PluginManager for CoffeeManager {
...
+                .map_err(|err| CoffeeError::new(1, &err.cause))?;
+            self.storage.store(&self.storage_info()).await?;
+            self.update_conf().await?;
+            return Ok(());
+        } else {
+            return Err(CoffeeError::new(
+                1,
+                &format!("plugin `{plugin}` is already not installed"),
+            ));
+        }
```
[the remark this answers is not in the document]
Comment (tareknaser360) [reply coffee_core/src/coffee.rs:256]: The function returns `Result<Value, CoffeeError>` 
How can I do this?

[the remark this answers is not in the document]
Comment (vincenzopalazzo) [reply coffee_core/src/coffee.rs:256]: `error!` is a macros under the `use coffee_lib::error;` and will create the CoffeeError for you

**Comment by vincenzopalazzo** [comment]
this need a rebase!

File: coffee_cmd/src/main.rs [hunk 4]
```diff
@@ -38,7 +38,13 @@ async fn main() -> Result<(), CoffeeError> {
...
             result
         }
-        CoffeeCommand::Remove => todo!(),
+        CoffeeCommand::Remove { plugin } => match coffee.remove(&plugin).await {
+            Ok(_) => {
+                term::success!("successfully uninstalled plugin {plugin}");
+                Ok(())
+            }
+            Err(err) => Err(err),
+        },
```
Comment (vincenzopalazzo) [inline coffee_cmd/src/main.rs:47]: can you follow the coffee_term style that I added in a previous PR?

File: coffee_core/src/coffee.rs [hunk 5]
```diff
@@ -234,6 +234,28 @@ impl PluginManager for CoffeeManager {
         Err(err)
     }
 
+    async fn remove(&mut self, plugin: &str) -> Result<Value, CoffeeError> {
+        debug!("removing plugin: {plugin}");
+        let values: &mut Vec<Plugin> = &mut self.config.plugins;
```
Comment (vincenzopalazzo) [inline coffee_core/src/coffee.rs:239]:
```suggestion
        let plugins = &mut self.config.plugins;
```

the compiler will infer the type alone

File: coffee_core/src/coffee.rs [hunk 6]
```diff
@@ -234,6 +234,28 @@ impl PluginManager for CoffeeManager {
...
+                .map_err(|err| CoffeeError::new(1, &err.cause))?;
+            self.storage.store(&self.storage_info()).await?;
+            self.update_conf().await?;
+            return Ok(serde_json::to_value(plugin).unwrap());
+        } else {
+            return Err(CoffeeError::new(
+                1,
+                &format!("plugin `{plugin}` is already not installed"),
+            ));
+        }
```
Comment (vincenzopalazzo) [inline coffee_core/src/coffee.rs:256]: Can we return a typed response? as I did in some previous method like `coffee_list` and `coffee_list_remotes`?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: coffee_core/src/coffee.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (tareknaser360) [reply coffee_core/src/coffee.rs:256]: Yes thanks. I rebased the branch on top of master

File: coffee_cmd/src/main.rs (hunk 4 above)
Comment (tareknaser360) [reply coffee_cmd/src/main.rs:47]: Done

File: coffee_core/src/coffee.rs (hunk 6 above)
Comment (tareknaser360) [reply coffee_core/src/coffee.rs:256]: Yes i created a new struct `CoffeeRemove` with one field `plugin`

File: coffee_lib/src/types/mod.rs [hunk 7]
```diff
@@ -3,6 +3,11 @@ use serde::{Deserialize, Serialize};
 
 use crate::plugin::Plugin;
 
+#[derive(Serialize, Deserialize)]
+pub struct CoffeeRemove {
+    pub plugin: Option<Plugin>,
```
Comment (vincenzopalazzo) [inline coffee_lib/src/types/mod.rs:8]:
```suggestion
    pub plugin: Plugin,
```

there is some case where the plugin in null in the json?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Ok small suggestion but we are there

File: coffee_lib/src/types/mod.rs (hunk 7 above)
Comment (tareknaser360) [reply coffee_lib/src/types/mod.rs:8]: No there isn't.
We already check if the plugin exists once.
I will fix this

**Review by vincenzopalazzo — APPROVED** [review]
Good job Thanks!

**Pull request merged**